src/lib/annotations/doc/DocHighlightDialog.js

```diff
@@ -40,13 +39,58 @@ class DocHighlightDialog extends AnnotationDialog {
             ]);
             annotatorUtil.showElement(highlightLabelEl);
 
-            // Only reposition if mouse is still hovering over the dialog
-            this.position();
+            // Only reposition if mouse is still hovering over the dialog and not mobile
+            if (!this.isMobile) {
+                this.position();
+            }
         }
 
         super.addAnnotation(annotation);
     }
 
+    /** @inheritdoc */
+    postAnnotation(textInput) {
+        const annotationTextEl = this.element.querySelector(constants.SELECTOR_ANNOTATION_TEXTAREA);
+        const text = textInput || annotationTextEl.value;
+        if (text.trim() === '') {
+            return;
+        }
+
+        // Convert from plain highlight to comment
+        const headerEl = this.isMobile ? this.element.querySelector('.bp-annotation-mobile-header') : null;
```

Comment:
This ternary operator seems unnecessary. Can you just do 
```
const mobileHeaderEl = this.element.querySelector('.bp-annotation-mobile-header');

if (mobileHeaderEl) {
    headerEl.classList.remove(CLASS_HIDDEN);
    this.element.classList.remove(constants.CLASS_ANNOTATION_PLAIN_HIGHLIGHT);
}
```
since the mobileHeaderEl won't exist if the user isn't on a mobile device?